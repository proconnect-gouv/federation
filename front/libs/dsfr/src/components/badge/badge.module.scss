.badge {
  height: fit-content;
  white-space: nowrap;
  width: fit-content;
}
