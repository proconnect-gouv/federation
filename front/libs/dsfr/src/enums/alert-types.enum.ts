/* istanbul ignore file */
export enum AlertTypes {
  ERROR = 'error',
  INFO = 'info',
  SUCCESS = 'success',
  WARNING = 'warning',
}
