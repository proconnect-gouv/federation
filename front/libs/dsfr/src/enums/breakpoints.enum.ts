/* istanbul ignore file */
export enum Breakpoints {
  XS = 576,
  SM = 768,
  MD = 992,
  LG = 1248,
}
