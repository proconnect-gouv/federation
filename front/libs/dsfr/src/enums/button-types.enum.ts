/* istanbul ignore file */
export enum ButtonTypes {
  SUBMIT = 'submit',
  BUTTON = 'button',
  RESET = 'reset',
}
