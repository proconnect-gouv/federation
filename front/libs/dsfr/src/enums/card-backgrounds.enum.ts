export enum CardBackgrounds {
  GREY = 'grey',
  NO_BACKGROUND = 'no-background',
  NO_BORDER = 'no-border',
  SHADOW = 'shadow',
}
