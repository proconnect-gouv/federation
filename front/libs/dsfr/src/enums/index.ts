/* istanbul ignore file */

// declarative file
export * from './alert-types.enum';
export * from './button-types.enum';
export * from './icon-placement.enum';
export * from './pagination.enum';
export * from './priorities.enum';
export * from './sizes.enum';
