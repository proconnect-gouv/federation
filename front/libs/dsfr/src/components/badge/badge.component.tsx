import classnames from 'classnames';
import React from 'react';
import { IconType } from 'react-icons';

import { Sizes } from '../../enums';
import styles from './badge.module.scss';

export interface BadgeComponentProps {
  colorName?: string;
  label: string;
  size?: Sizes;
  icon?: IconType | undefined;
  iconSize?: number;
  dataTestId?: string;
  noIcon?: boolean;
}

export const BadgeComponent: React.FC<BadgeComponentProps> = React.memo(
  ({ colorName, dataTestId, icon: Icon, iconSize, label, noIcon, size }: BadgeComponentProps) => (
    <div
      className={classnames(styles.badge, `fr-badge fr-badge--${colorName} fr-badge--${size}`, {
        // DSFR classname
        // eslint-disable-next-line @typescript-eslint/naming-convention
        'fr-badge--no-icon': !!Icon || noIcon,
      })}
      data-testid={dataTestId}>
      {!noIcon && Icon && <Icon className="fr-mr-1w" size={iconSize} />}
      <b>{label}</b>
    </div>
  ),
);

BadgeComponent.defaultProps = {
  colorName: 'grey',
  dataTestId: undefined,
  icon: undefined,
  iconSize: 18,
  noIcon: false,
  size: Sizes.MEDIUM,
};

BadgeComponent.displayName = 'BadgeComponent';
