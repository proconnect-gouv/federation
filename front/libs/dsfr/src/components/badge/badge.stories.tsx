import { ComponentMeta } from '@storybook/react';
import { RiAccountCircleFill as UserIcon } from 'react-icons/ri';

import { ServiceProviderStatusColors } from '@fc/partners';

import { Sizes } from '../../enums';
import { BadgeComponent, BadgeComponentProps } from './badge.component';

export default {
  component: BadgeComponent,
  title: 'DSFR/components/badge/BadgeComponent',
} as ComponentMeta<typeof BadgeComponent>;

export const Default = (args: BadgeComponentProps) => <BadgeComponent {...args} />;

export const WithCustomIcon = (args: BadgeComponentProps) => <BadgeComponent {...args} />;

Default.args = {
  label: 'badge',
  noIcon: false,
  size: Sizes.MEDIUM,
} as BadgeComponentProps;

WithCustomIcon.args = {
  colorName: ServiceProviderStatusColors.REVIEW_WAITING_CLIENT_FEEDBACK,
  icon: UserIcon,
  label: 'badge',
  noIcon: false,
  size: Sizes.MEDIUM,
} as BadgeComponentProps;
