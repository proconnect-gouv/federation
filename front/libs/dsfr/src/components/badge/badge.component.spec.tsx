import { render } from '@testing-library/react';
import { RiAccountCircleFill as UserIcon } from 'react-icons/ri';

import { Sizes } from '../../enums';
import { BadgeComponent } from './badge.component';

describe('BadgeComponent', () => {
  it('should match the snapshot with default value', () => {
    // When
    const { container } = render(<BadgeComponent label="label" />);

    // Then
    expect(container).toMatchSnapshot();
  });

  it('should match the snapshot with default value and custom icon', () => {
    // When
    const { container } = render(<BadgeComponent Icon={UserIcon} label="label" />);

    // Then
    expect(container).toMatchSnapshot();
  });

  it('should have class fr-badge--custom-color if colorName is "custom-color"', () => {
    // When
    const { getByTestId } = render(
      <BadgeComponent colorName="custom-color" dataTestId="BadgeComponent" label="label" />,
    );
    const element = getByTestId('BadgeComponent');

    // Then
    expect(element).toHaveClass('fr-badge--custom-color');
  });

  it('should have class fr-badge--sm if size is "sm"', () => {
    // When
    const { getByTestId } = render(
      <BadgeComponent dataTestId="BadgeComponent" label="label" size={Sizes.SMALL} />,
    );
    const element = getByTestId('BadgeComponent');

    // Then
    expect(element).toHaveClass('fr-badge--sm');
  });

  it('should have class fr-badge--no-icon if noIcon is set to true', () => {
    // When
    const { getByTestId } = render(
      <BadgeComponent noIcon dataTestId="BadgeComponent" label="label" />,
    );
    const element = getByTestId('BadgeComponent');

    // Then
    expect(element).toHaveClass('fr-badge--no-icon');
  });

  it('should have class fr-badge--no-icon if we define a custom icon', () => {
    // When
    const { getByTestId } = render(
      <BadgeComponent dataTestId="BadgeComponent" Icon={UserIcon} label="label" />,
    );
    const element = getByTestId('BadgeComponent');

    // Then
    expect(element).toHaveClass('fr-badge--no-icon');
  });

  it('should have element <Icon /> if we define a custom icon and noIcon value set to false', () => {
    // Given
    jest.mocked(UserIcon);

    // When
    render(<BadgeComponent Icon={UserIcon} label="label" />);

    // Then
    expect(UserIcon).toHaveBeenCalledOnce();
  });

  it('should not have element <Icon /> even if we define a custom icon with noIcon value at true', () => {
    // Given
    jest.mocked(UserIcon);

    // When
    render(<BadgeComponent noIcon Icon={UserIcon} label="label" />);

    // Then
    expect(UserIcon).toHaveBeenCalledTimes(0);
  });
});
