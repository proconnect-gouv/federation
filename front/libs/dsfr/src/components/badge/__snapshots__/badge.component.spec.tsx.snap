// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BadgeComponent should match the snapshot with default value 1`] = `
<div>
  <div
    class="badge fr-badge fr-badge--grey fr-badge--md"
  >
    <b>
      label
    </b>
  </div>
</div>
`;

exports[`BadgeComponent should match the snapshot with default value and custom icon 1`] = `
<div>
  <div
    class="badge fr-badge fr-badge--grey fr-badge--md fr-badge--no-icon"
  >
    <div
      class="fr-mr-1w"
      size="18"
    >
      RiAccountCircleFill
    </div>
    <b>
      label
    </b>
  </div>
</div>
`;
